propnet/core/tests/test_graph.py

```diff
@@ -45,26 +88,107 @@ def testEvaluateMethod1(self):
+
+    def testEvaluateMethod2(self):
+        """
+        Graph has two materials on it: mat1 & mat2
+            mat1 has nondegenerate properties relative permittivity and relative permeability
+            mat2 has nondegenerate properties relative permittivity and relative permeability
+        Determines if TestGraph1 is correctly evaluated using the evaluate method.
+        We expect 4 refractive_index properties to be calculated as the following:
+            sqrt(3), sqrt(5), sqrt(6), sqrt(10)
+        We expect each material graph to have 2 include refractive_index nodes.
+        We expect each property jointly calculated to have a joint_material tag element.
+        """
+        # Setup
+        propnet = Propnet()
+        mat1 = Material()
+        mat2 = Material()
+        mat1.add_property(Symbol(SymbolType['relative_permeability'], 1, None))
+        mat2.add_property(Symbol(SymbolType['relative_permeability'], 2, None))
+        mat1.add_property(Symbol(SymbolType['relative_permittivity'], 3, None))
+        mat2.add_property(Symbol(SymbolType['relative_permittivity'], 5, None))
+        propnet.add_material(mat1)
+        propnet.add_material(mat2)
+
+        propnet.evaluate()
+
+        # Expected propnet outputs
+        s_outputs = []
+        s_outputs.append(Symbol(SymbolType['relative_permeability'], 1, None))
+        s_outputs.append(Symbol(SymbolType['relative_permeability'], 2, None))
+        s_outputs.append(Symbol(SymbolType['relative_permittivity'], 3, None))
+        s_outputs.append(Symbol(SymbolType['relative_permittivity'], 5, None))
+        s_outputs.append(Symbol(SymbolType['refractive_index'], 3 ** 0.5, None))
+        s_outputs.append(Symbol(SymbolType['refractive_index'], 5 ** 0.5, None))
+        s_outputs.append(Symbol(SymbolType['refractive_index'], 6 ** 0.5, None))
+        s_outputs.append(Symbol(SymbolType['refractive_index'], 10 ** 0.5, None))
+
+        m_outputs = [mat1, mat2]
+
+        # Expected mat_1 outputs
+        m1_s_outputs = []
+        m1_s_outputs.append(Symbol(SymbolType['relative_permeability'], 1, None))
+        m1_s_outputs.append(Symbol(SymbolType['relative_permittivity'], 3, None))
+        m1_s_outputs.append(Symbol(SymbolType['refractive_index'], 3 ** 0.5, None))
+
+        #Expected mat_2 outputs
+        m2_s_outputs = []
+        m2_s_outputs.append(Symbol(SymbolType['relative_permeability'], 2, None))
+        m2_s_outputs.append(Symbol(SymbolType['relative_permittivity'], 5, None))
+        m2_s_outputs.append(Symbol(SymbolType['refractive_index'], 10 ** 0.5, None))
+
+        st_outputs = []
+        st_outputs.append(SymbolType['relative_permeability'])
+        st_outputs.append(SymbolType['relative_permittivity'])
+        st_outputs.append(SymbolType['refractive_index'])
+
+        # Test
+        if not GraphTest.checkGraphSymbols(mat1.graph, m1_s_outputs, 'Symbol'):
+            raise ValueError('Graph symbol structure incorrect.')
```

Comment:
This isn't really a test, I don't think tests should raise anything -- if you are expecting to raise an exception from a test, you should use `self.assertRaises()` -- there's a whole list of these, e.g. `assertEqual`, `assertTrue`, etc.

Reply:
Ok will change raise to assert and re-push